index.js

```diff
@@ -940,7 +945,7 @@ module.exports = function (log, indexesPath) {
 
   function executeOperation(operation, cb) {
     updateCacheWithLog()
-    if (bitsetCache.has(operation)) return cb(bitsetCache.get(operation))
+    if (bitsetCache.has(operation)) return cb(...bitsetCache.get(operation))
```

Comment:
One thing I'm concerned is whether the `filters` Map will be reused, because when there are AND or OR operations, the 1st `filters` Map gets *mutated*, and in this `bitsetCache` we are storing that mutated Map.

It could be we're totally safe, but I'm just trying to think if we have that guarantee or not.

Reply:
I had to double check but I think we should be safe. Basically as long as the log does not change (which the cache already has a check for), then for a given operation the bitset + filters should be exactly the same. The only thing that could change is the seq or limit and for those we can reuse the bitset and filters as they are not mutated in `getMessagesFromBitsetSlice`.

Reply:
I take that back. If you do have a query like `or(and(type('post'), author('a')), and(type('vote'), author('a')))` then that could potentially have a problem as you describe. I tested just creating a new map when merging and could not measure any perf diff, so better safe than sorry.